instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/SqsAccess.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.awssdk.v2_2;
+
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.javaagent.tooling.muzzle.NoMuzzle;
+import software.amazon.awssdk.core.SdkRequest;
+import software.amazon.awssdk.core.interceptor.Context;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+
+// helper class for calling methods that use sqs types in SqsImpl
+// if SqsImpl is not present these methods are no op
+final class SqsAccess {
+  private static final boolean enabled = isSqsImplPresent();
+
+  private static boolean isSqsImplPresent() {
+    try {
+      // for library instrumentation SqsImpl is always available
+      // for javaagent instrumentation SqsImpl is available only when SqsInstrumentationModule was
+      // successfully applied (muzzle passed)
+      // using package name here because library instrumentation classes are relocated when embedded
+      // in the agent
+      Class.forName(SqsAccess.class.getPackage().getName() + ".SqsImpl");
```

Comment:
BTW, is getPackage really safe here? Doesn't it sometimes return `null`? Maybe we should use lastIndexOf('.') on the class name like https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/v1.27.0/javaagent-bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/AgentClassLoader.java#L234-L237

Reply:
You are correct it can return null in class loaders that don't call `definePackage`. An alternative to using substring would be to use `SqsAccess.class.getName().replace("SqsAccess", "SqsImpl")`.

Reply:
Let's try just that:

```suggestion
      Class.forName(SqsAccess.class.getName().replace(".SqsAccess", ".SqsImpl"));
```